### PR TITLE
feat: Introduce document thrown exceptions

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
This checks rule [8@104](https://tics.tiobe.com/viewerCS/?ID=107&CSTD=Rule) of the Philips C# Coding Standard.

### CODE REVIEW

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentThrownExceptionsAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.ThrowStatement);
+		}
+
+		private void Analyze(SyntaxNodeAnalysisContext context)
+		{
+			var throwStatement = (ThrowStatementSyntax)context.Node;
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentThrownExceptionsAnalyzer.cs:37]: We don't want to document a re-throw?

File: Philips.CodeAnalysis.Test/Maintainability/Documentation/DocumentThrownExceptionsAnalyzerTest.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+{
+	[TestClass]
+	public class DocumentThrownExceptionsAnalyzerTest : DiagnosticVerifier
+	{
+		protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+		{
+			return new DocumentThrownExceptionsAnalyzer();
+		}
+
+		private const string CorrectNoThrow = @"
```
Comment (bcollamore) [inline Philips.CodeAnalysis.Test/Maintainability/Documentation/DocumentThrownExceptionsAnalyzerTest.cs:23]: Add a test for throw expressions

**Review by bcollamore — APPROVED** [review]
We should try to include CodeFixers wherever reasonable, in order to help productivity/adoption of the rules.

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentThrownExceptionsAnalyzer.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentThrownExceptionsAnalyzer.cs:37]: Oops, added

**Review by bcollamore — DISMISSED** [review]

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentThrownExceptionsAnalyzer.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentThrownExceptionsAnalyzer.cs:37]: Iff I push my changes ;-)

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Philips.CodeAnalysis.MaintainabilityAnalyzers.md [hunk 3]
```diff
@@ -69,3 +69,5 @@
 | PH2116  | Avoid ArrayList                              | Usage of Arraylist is discouraged by Microsoft for performance reasons, use List<T> instead. |
 | PH2117  | Avoid Unnecessary Where()                    | Move the predicate of the Where clause into the Any(), Count(), First(), Last(), or Single() clause |
 | PH2118  | Avoid inline magic numbers                   | Avoid inline magic number, define them as constant or include in an enumeration instead. |
+| PH2120  | Document thrown exceptions                   | Be clear to your callers what exception can be thrown from your method by mentioning each of them in an <exception> element in the documentation of the method |
```
[the remark this answers is not in the document]
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Philips.CodeAnalysis.MaintainabilityAnalyzers.md:72]: PH2119 will come in with #235

**Review by bcollamore — APPROVED** [review]

**Pull request merged**